### PR TITLE
Add Travis-CI simulation using Docker

### PR DESCRIPTION
**Pull request by briandealwis**
This patch adds support for simulating a Travis build using Docker.  It's useful for debugging why a test may be failing.  It's not a 100% simulation — I think `tightvncserver` brings in a window manager, so we don't use _metacity_ as we are in the Travis build — but it works well enough.

### CODE REVIEW

**Comment by akerekes** [comment]
Awesome! :fireworks:

**Comment by chanseokoh** [comment]
Cool!

File: README.md [hunk 1]
```diff
@@ -247,3 +247,12 @@ the version should be specified as `"0.0.0"` to indicate that the
 current version found should be used.  Unlike the `.tpd` file,
 the identifiers are not p2 identifiers, and so features do not
 require the `.feature.group` suffix.
+
+# Travis-CI
+
+This project is automatically built and tested thanks to [Travis-CI](https://travis-ci.org/GoogleCloudPlatform/gcloud-eclipse-tools/).
```
Comment (elharo) [inline README.md]: thanks to --> by

File: README.md [hunk 2]
```diff
@@ -247,3 +247,12 @@ the version should be specified as `"0.0.0"` to indicate that the
...
+
+## Simulating and Debugging Travis-CI builds
+ 
+We have a Docker setup to simulate the Travis-CI build, found in [`build/travis-debug`](build/travis-debug).
```
Comment (elharo) [inline README.md]: A Docker setup that simulates the Travis-CI build is found in

File: build/travis-debug/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Simulate our Maven build on Travis-CI
+# WORKDIR ends up as /home/travis/build
+# Source directory should be mounted here
+
+FROM quay.io/travisci/travis-jvm
```
Comment (elharo) [inline build/travis-debug/Dockerfile]: I have to ask about the security of this image. Who provides it? How is it vetted?

Comment (briandealwis) [inline build/travis-debug/Dockerfile]: That's the [Travis-provided image that they use in production](https://docs.travis-ci.com/user/common-build-problems/#Running-a-Container-Based-Docker-Image-Locally).

Comment (elharo) [inline build/travis-debug/Dockerfile]: How do we verify that this is the Travis provided image? Is there some sort of signature checking on the binary? quay.io is a third party vendor.

Comment (briandealwis) [inline build/travis-debug/Dockerfile]: I don't know of any way to verify.  We could specify an [explicit tag](https://quay.io/repository/travisci/travis-jvm?tab=tags).  Looking at `docker history`, I don't see a referenced image.

We could re-build this against a standard image like the [`maven` images](https://hub.docker.com/_/maven/).

**Comment by briandealwis** [comment]
Travis published their [converter for `.travis.yml` &rarr; build.sh](https://github.com/travis-ci/travis-build).

**Comment by elharo** [comment]
FYI

http://security.stackexchange.com/questions/88790/is-it-safe-to-trust-a-docker-container

The short answer is no, we can't trust a container.

**Pull request closed without merging**